core/spec/lib/spree/migration_helpers_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+
+RSpec.describe Spree::MigrationHelpers do
+  let(:helper) do
+    double.as_null_object.tap do |object|
+      object.extend(described_class)
+      object.extend(ActiveRecord::ConnectionAdapters::SchemaStatements)
+    end
+  end
+
+  subject { helper.safe_add_index(double, double) }
+
+  # regression test
+  describe "#safe_add_index" do
+    context "when the column exists" do
+      context "and the index does" do
+        it "is compatible with Ruby 3.0" do
```

Comment:
Not blocking, but I wouldn't be specific about the version under test in the `it` description. That's something that is controlled by the CI matrix; for example, if this spec is run in the CI with Ruby 2.7, it will still pass. My suggestion is to just document the behavior, like `it "add the index correctly" do`, if it will fail when testing against Ruby 3.0, we'll get to know by the right CI step. Thoughts?

Reply:
This is a good point @kennyadsl. Though I think `add the index correctly` is misleading, as we aren't testing that it works fully. Due to it's incomplete setup, the method actually raises other errors related to the migration. So we are specifically only testing argument compatibility.

WDYT about:
- `it "passes compatible arguments to index_exists?" do`; and
- `it "passes compatible arguments to add_index" do`

Reply:
Both Make sense, that's what I wanted to propose 👍. Go with the one you like the most.